### ISSUE TITLE
Let user decide what's the default

### ISSUE THREAD

**Issue by pascalwhoop** [meetDeveloper/JavaScript-Switcher#2]
Hey. 
What do you think about letting the user decide what is the default? Would that be hard to implement? I got no idea, never wrote a firefox extension. But you could have a toggle "default" with on and off. So everyone can pick their favorite.

**Comment by pabgan** [comment]
Yes, please. I really need a way to set disabled as default. Isn't there a hidden config somewhere?

**Comment by meetDeveloper** [comment]
@pascalwhoop So Sorry, I never starred the repository so never got notified of your comment, I will implement that soon, in few day. Really Sorry for so long wait. I would do that very soon. It is not at all hard.

**Comment by meetDeveloper** [comment]
@pabgan I will implement it in few days. Sorry for so late response.

**Comment by meetDeveloper** [comment]
If you need anything else feel free to open another issue. I would be more than happy to implement new features.

**Comment by pabgan** [comment]
Hi @meetDeveloper, thanks for the interest. Is that ready? I did not see any commit or updates in the extension. The extension I am using right now is driving me crazy and I would very much like to use yours.

**Comment by meetDeveloper** [comment]
@pabgan Sorry Brother, I am getting on it right now, can you tell me what all thing in your extension is driving you crazy, I could better understand the situation that way.

**Comment by pabgan** [comment]
No worries, thank you for your work.

The one I am using now is 
https://addons.mozilla.org/es/firefox/addon/disable-javascript/
I have configured it to JS off by default and disable JS by tab, not by 
domain.

Is driving me crazy two things most of all:
1. It does not have memory or a whitelist for websites where I do want 
JS always.
2. The keyboard shortcut to enable/disable JS does not work 90% of the 
time, not sure why.

What I would like to do is:
1. Navigate without JS to the extent possible.
2. When I really need to use a website and it does not work without JS 
I want to be able to enable it temporarily with a shortcut, (a button 
as an extra does not harm)	.
3. If I need to use the non-functioning-without-js website regularly, 
like the my work's Jira application, then I want to add the domain to a 
whitelist, with a shortcut too if that is possible.

Thanks again, bro.

**Comment by meetDeveloper** [comment]
@pabgan Sorry, for late reply, I missed a notification for this message, I will look into it and improve my extension, I currently do job, and get little time to work on these things. But will surely take up next weekend and improve it.

**Comment by resynth1943** [comment]
@pabgan You may prefer something like [NoScript](https://addons.mozilla.org/en-GB/firefox/addon/noscript/), which provides more advanced control over the scripts your browser loads.

**Comment by pabgan** [comment]
@resynth1943 , thanks for the suggestion. I actually used that one for a long while, but finally moved on, to other options. I don't recall the reason exactly, might be that it is missing a "temporarily allow this tab" or something like this. I have tried a few extensions and all of them miss something that the others have.

I think I am going to have a little bit more free time next year, and I plan to improve any of those, whichever seems easier.

**Comment by meetDeveloper** [comment]
@pabgan Can you improve this one?

**Comment by pabgan** [comment]
I will definitely take a look at it.

**Comment by ghost** [comment]
This addon is exactly what I was looking for. But enabling Javascript as the default setting is a bad idea, even if most people want it, because most people do not know how powerful and therefore dangerous javascript is. Please change the default setting to disabled or let the user decide as posted above. We understand, that you have a real life and a job, but please release an update as promised!

**Comment by phil294** [comment]
fyi the relevant code is setting `let whitelist_js = false` in `background.js`
